### PR TITLE
Upgrade gradle 4.10.2 -> 6.5.1

### PR DESCRIPTION
**Pull request by doncem**
In order to #226 gradle needs to be 5.6 or 6.3.

This has been lying around locally for some time now. Pushing along `dependencyUpdates` task fix which started to work locally and helped me with stuff.

Solving some deprecations too.

Tested locally - snapshot got built and project loaded everything up. `bintrayUpload` still worked as well - got 401 to the point where need to upload jars etc.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -92,20 +86,17 @@ allprojects {
...
-      def isNonStable = { String version ->
-        ['alpha', 'beta', 'rc', 'cr', 'm', 'preview', 'b', 'ea'].any { qualifier ->
-          version ==~ /(?i).*[.-]\$qualifier[.\\d-+]*/
-        }
-      }
-      if (isNonStable(candidate.version) && !isNonStable(currentVersion)) {
-        selection.reject('Release candidate')
-      }
+  // before committing a change, make sure task still works
+  dependencyUpdates {
```
Comment (timja) [inline build.gradle:90]: this seems to be duplicated in the file?

File: gradle/wrapper/gradle-wrapper.properties [hunk 2]
```diff
@@ -1,6 +1,5 @@
-#Mon Dec 17 11:02:30 GMT 2018
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-6.5-all.zip
```
Comment (timja) [inline gradle/wrapper/gradle-wrapper.properties:3]:
```suggestion
distributionUrl=https\://services.gradle.org/distributions/gradle-6.5.1-all.zip
```

File: build.gradle (hunk 1 above)
Comment (doncem) [reply build.gradle:90]: back in a day, i think used to not work if only inside `allprojects` section

Comment (doncem) [reply build.gradle:90]: worked when removed :shrug:

File: build.gradle [hunk 3]
```diff
@@ -43,7 +31,8 @@ allprojects {
   }
 
   pmd {
-    toolVersion = "6.16.0"
+    toolVersion = "6.25.0"
+    incrementalAnalysis = true
```
Comment (timja) [inline build.gradle:35]: o didn't know they got this working, nice

Comment (doncem) [reply build.gradle:36]: since gradle 5 point something :upside_down_face:

**Pull request merged**